src/raftstore/store/peer.rs

```diff
@@ -784,6 +818,7 @@ impl Peer {
                 }
                 StateRole::Follower => {
                     self.leader_lease.expire();
+                    self.deregister_read_delegate();
```

Comment:
do we need to deregister when the peer is removed?

Reply:
You are right. Merge will destory leader which leaves an orphan read delegate.